default.nix

```diff
@@ -187,7 +187,11 @@ lib.makeScope pkgs.newScope (self: {
           lockPkgs = builtins.listToAttrs (
             builtins.map
               (
-                pkgMeta: rec {
+                pkgMeta:
+                if pkgMeta.name == "setuptools" then {
+                  name = pkgMeta.name;
+                  value = super.setuptools;
```

Comment:
This should probably be extended to cover all the build systems we support...

Reply:
Yeah, that makes sense. I'd like to see if I can get to the bottom of the reason why the infinite recursion is happening before doing that though.

One downside to this approach is that the version of whatever the build system is always pinned to whatever version/commit nixpkgs is pinned to. For setuptools that probably doesn't matter much, but for the newer build systems I'm not so sure.